packages/ember-data/tests/integration/serializers/embedded-records-mixin-test.js

```diff
@@ -1165,6 +1165,63 @@ test("extractSingle with polymorphic hasMany", function() {
 
 });
 
+test("extractSingle with polymorphic hasMany and custom primary key", function() {
+  SuperVillain.reopen({
+    secretWeapons: DS.hasMany("secretWeapon", { polymorphic: true })
+  });
+
+  env.registry.register('adapter:super-villain', DS.ActiveModelAdapter);
+  env.registry.register('serializer:light-saber', DS.ActiveModelSerializer.extend({
+    primaryKey: 'custom'
+  }));
+  env.registry.register('serializer:super-villain', DS.ActiveModelSerializer.extend(DS.EmbeddedRecordsMixin, {
+    attrs: {
+      secretWeapons: { embedded: 'always' }
+    }
+  }));
+  var serializer = env.container.lookup("serializer:super-villain");
+
+  var json_hash = {
+    super_villain: {
+      id: "1",
+      first_name: "Tom",
+      last_name: "Dale",
+      secret_weapons: [
+        {
+          custom: "1",
+          type: "LightSaber",
+          name: "Tom's LightSaber",
+          color: "Red"
+        },
+        {
+          id: "1",
```

Comment:
I could add a comment since I guess this would make reading the test easier ... c/d?